core/src/main/scala/org/json4s/reflect/Reflector.scala

```diff
@@ -124,16 +132,21 @@ object Reflector {
           case None => ctors.map(new Executable(_, false))
         }
       }
+
       val constructorDescriptors = createConstructorDescriptors(ccs)
-      companion = findCompanion(checkCompanionMapping = false)
-      val applyMethods: scala.Array[Method] = companion match {
-        case Some(singletonDescriptor) =>
-          singletonDescriptor.instance.getClass.getMethods.filter { method => method.getName == "apply" && method.getReturnType == er }
-        case None => scala.Array[Method]()
+      val all = {
```

Comment:
Would you prefer that this be refactored into a separate function?

Reply:
I think it's fine for now.